### PR TITLE
Jsonize autodoc couch

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Infrastructure "Jsonize autodoc couch"```

#### Purpose of change
Code was checking for the furniture id to find the couch, now it checks for a flag

#### Describe the solution
Check for AUTODOC_COUCH flag instead of f_autodoc_couch furniture id
Check for AUTODOC flag instead of f_autodoc furniture id

#### Testing
Give autodoc iexamine to armchair
Give AUTODOC to armchair
Give AUTODOC_COUCH to domestic plant
Stand on plant
Examine armchair
Start installing CBMs

#### Additional context
This would make moding autodocs easier

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
I'm not against jsonizing, of course, but autodoc couch isn't any couch or sofa you find in nearest house, it has several specific features (like hand wraps or integrated machinery or complex electronics connected to the autodoc) in it. I'm not sure if it would be possible to mod some other furniture to simulate it being the proper autodoc couch.

**Comment by SirPendrak** [comment]
Does it mean we can hope for mobile vechicle versions of autodoc in the future?

**Comment by Fris0uman** [comment]
No, vehicle part are something else, and would require a whole bunch of new code. But you can make a craftable item that deploys into an autodoc if you want. It might work with just one furniture that has both the flag and the examine action, and if it doesn't you would need two furniture: a console and a couch.

File: src/iexamine.cpp [hunk 1]
```diff
@@ -4263,8 +4262,7 @@ static Character &operator_present( Character &p, const tripoint &autodoc_loc,
                                     Character &null_patient )
 {
     for( const auto &loc : g->m.points_in_radius( autodoc_loc, 1 ) ) {
-        const furn_str_id couch( "f_autodoc_couch" );
-        if( g->m.furn( loc ) != couch ) {
+        if( !g->m.furn( loc )->has_flag("AUTODOC_COUCH")) {
```
Comment (BevapDin) [inline src/iexamine.cpp]: Please use `map::has_flag_furn` or even just `map::has_flag` (which includes flags from terrain).

**Comment by BevapDin** [comment]
`f_autodoc_couch` is used at other places:

"activity_handlers.cpp"
"monattach.cpp"
"monmove.cpp" (twice).

Also consider removing the global variable `f_autodoc_couch` (declared in "mapdata.h").

File: src/map.h [hunk 2]
```diff
@@ -1804,9 +1804,9 @@ class map
         std::list<item_location> get_active_items_in_radius( const tripoint &center, int radius,
                 special_item_type type ) const;
 
-        /**returns positions of furnitures matching target in the specified radius*/
-        std::list<tripoint> find_furnitures_in_radius( const tripoint &center, size_t radius,
-                furn_id target,
+        /**returns positions of furnitures with matching flag in the specified radius*/
+        std::list<tripoint> find_furnitures_with_flag_in_radius( const tripoint &center, size_t radius,
+                std::string flag,
```
Comment (BevapDin) [inline src/map.h]:
```suggestion
                const std::string &flag,
```

File: src/monmove.cpp [hunk 3]
```diff
@@ -62,6 +62,8 @@ static const efftype_id effect_harnessed( "harnessed" );
 
 static const species_id ZOMBIE( "ZOMBIE" );
 
+static const std::string flag_AUTODOC_COUCH( "AUTODOC_COUCH" );
```
Comment (BevapDin) [inline src/monmove.cpp]: Out of curiosity: why is some code using a string literal (e.g. in `mattack::nurse_operate`) and some code a global constant?

Comment (Fris0uman) [reply src/monmove.cpp]: I used the constant when the string was used more than once, is that wrong?

Comment (BevapDin) [reply src/monmove.cpp]: It's definitively not wrong. It's just inconsistent and makes people wonder if there's a meaning to it. Especially as the name of the constant is the same as the string literal. Reading `flag_AUTODOC_COUCH` doesn't convey more information than "AUTODOC_COUCH".

Comment (Fris0uman) [reply src/monmove.cpp]: So I should use the constant everywhere?

**Pull request merged**